File: Grunt/Grunt/Models/HaloInfinite/Medal.cs
// <copyright file="Medal.cs" company="Den Delimarsky">
// Developed by Den Delimarsky.
// Den Delimarsky licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.
// The underlying API powering Grunt is managed by 343 Industries and Microsoft. This wrapper is not endorsed by 343 Industries or Microsoft.
// </copyright>

namespace OpenSpartan.Grunt.Models.HaloInfinite
{
    /// <summary>
    /// Halo Infinite match performance medal.
    /// </summary>
    [IsAutomaticallySerializable]
    public class Medal
    {
        /// <summary>
        /// Gets or sets the medal name ID.
        /// </summary>
        public long NameId { get; set; }

        /// <summary>
        /// Gets or sets the medal count.
        /// </summary>
        public int Count { get; set; }

        /// <summary>
        /// Gets or sets the amount of personal score awarded.
        /// </summary>
        public int TotalPersonalScoreAwarded { get; set; }

        /// <summary>
        /// Gets or sets the medal name. Includes translated strings.
        /// </summary>
        public DisplayString? Name { get; set; }

        /// <summary>
        /// Gets or sets the medal description. Includes translated strings.
        /// </summary>
        public DisplayString? Description { get; set; }

        /// <summary>
        /// Gets or sets the sprite index. The sprite index is zero-based, starting with 0 on the sprite sheet. Sprite sheets are generally 16 by 16.
        /// </summary>
        public int SpriteIndex { get; set; }

        /// <summary>
        /// Gets or sets the sorting weight.
        /// </summary>
        public int SortingWeight { get; set; }

        /// <summary>
        /// Gets or sets the difficulty index.
        /// </summary>
        public int DifficultyIndex { get; set; }

        /// <summary>
        /// Gets or sets the type index.
        /// </summary>
        public int TypeIndex { get; set; }

        /// <summary>
        /// Gets or sets the personal score.
        /// </summary>
        public int PersonalScore { get; set; }
    }
}

File: src/node/src/models/halo-infinite/misc/medal.ts
import type { DisplayString } from '../economy/inventory';

/**
 * Medal definition.
 */
export interface Medal {
  /** Medal name identifier */
  NameId?: number;
  /** Number of times earned */
  Count?: number;
  /** Total personal score awarded */
  TotalPersonalScoreAwarded?: number;
  /** Display name */
  Name?: DisplayString;
  /** Description */
  Description?: DisplayString;
  /** Sprite index on the sprite sheet (zero-based) */
  SpriteIndex?: number;
  /** Sorting weight */
  SortingWeight?: number;
  /** Difficulty index, mapped to the difficulties property in medal metadata */
  DifficultyIndex?: number;
  /** Type index, mapped to the types property in medal metadata */
  TypeIndex?: number;
  /** Personal score */
  PersonalScore?: number;
}

/**
 * Medal metadata collection.
 */
export interface MedalMetadata {
  /** List of medals */
  Medals?: Medal[];
  /** Sprite sheet information */
  SpriteSheet?: Sprite;
}

/**
 * Sprite sheet information.
 */
export interface Sprite {
  /** Path to sprite sheet image */
  Path?: string;
  /** Number of columns */
  Columns?: number;
  /** Sprite size */
  Size?: number;
}
